### PR TITLE
[Backport to 2.x]Catalog to Datasource changes (#1027)

### PR DESCRIPTION
**Pull request by vamsi-amazon**
Signed-off-by: vamsi-amazon <reddyvam@amazon.com>
(cherry picked from commit 3e30379903ba596b8565e680af69073a16cdf940)

### Description
[Describe what this change achieves]
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by joshuali925 — APPROVED** [review]
should 2.4 be merged into 2.x?

**Comment by vamsi-amazon** [comment]
> should 2.4 be merged into 2.x?

Yes, But I am not sure if there are any pending PRs which are not merged into 2.x

**Comment by vamsi-amazon** [comment]
> > should 2.4 be merged into 2.x?
> 
> Yes, But I am not sure if there are any pending PRs which are not merged into 2.x

This is the only one I could find.

**Comment by joshuali925** [comment]
ok i checked the other missing one is https://github.com/opensearch-project/sql/pull/1050, should be good

File: docs/user/general/identifiers.rst [hunk 1]
```diff
@@ -231,7 +231,7 @@ tableName = ``http_requests_total``.
 2. ``logs.12.13.1``
 
 
-catalogName = ``@opensearch`` [Resolves to default @opensearch connector since [my_prometheus] is the only catalog configured name.]
+datasourceName = ``@opensearch`` [Resolves to default @opensearch connector since [my_prometheus] is the only catalog configured name.]
```
Comment (YANG-DB) [inline docs/user/general/identifiers.rst:234]: missed one here ...

File: docs/user/general/identifiers.rst [hunk 2]
```diff
@@ -241,23 +241,23 @@ tableName = ``logs.12.13.1``.
...
+datasourceName = ``@opensearch`` [Resolves to default @opensearch connector since [my_prometheus] is the only datasource configured name.]
 
 schemaName = ``default`` [No supported schema found, so default to `default`].
 
 tableName = ``prometheus.http_requests_total``.
 
 5. ``prometheus.default.http_requests_total.1.2.3``
 
-catalogName = ``@opensearch`` [Resolves to default @opensearch connector since [my_prometheus] is the only catalog configured name.]
+datasourceName = ``@opensearch`` [Resolves to default @opensearch connector since [my_prometheus] is the only catalog configured name.]
```
Comment (YANG-DB) [inline docs/user/general/identifiers.rst:260]: missed one here

File: doctest/build.gradle [hunk 3]
```diff
@@ -104,7 +104,7 @@ String mlCommonsPlugin = 'opensearch-ml'
 
 testClusters {
     docTestCluster {
-        keystore 'plugins.query.federation.catalog.config', new File("$projectDir/catalog", 'catalog.json')
+        keystore 'plugins.query.federation.datasources.config', new File("$projectDir/catalog", 'catalog.json')
```
Comment (YANG-DB) [inline doctest/build.gradle:107]: why not change the json file name also into datasource.json ?

File: integ-test/src/test/java/org/opensearch/sql/ppl/ShowCatalogsCommandIT.java [hunk 4]
```diff
@@ -20,25 +20,25 @@ public class ShowCatalogsCommandIT extends PPLIntegTestCase {
 
   @Test
   public void testShowCatalogsCommands() throws IOException {
```
Comment (YANG-DB) [inline integ-test/src/test/java/org/opensearch/sql/ppl/ShowCatalogsCommandIT.java:22]: should be testShowDatasourcesCommands()...

File: integ-test/src/test/java/org/opensearch/sql/ppl/ShowCatalogsCommandIT.java [hunk 5]
```diff
@@ -20,25 +20,25 @@ public class ShowCatalogsCommandIT extends PPLIntegTestCase {
...
     verifyColumn(
         result,
-        columnName("CATALOG_NAME"),
+        columnName("DATASOURCE_NAME"),
         columnName("CONNECTOR_TYPE")
     );
   }
 
   @Test
   public void testShowCatalogsCommandsWithWhereClause() throws IOException {
```
Comment (YANG-DB) [inline integ-test/src/test/java/org/opensearch/sql/ppl/ShowCatalogsCommandIT.java:35]: change test name here also...

File: integ-test/build.gradle [hunk 6]
```diff
@@ -111,7 +111,7 @@ testClusters.all {
 
 testClusters.integTest {
     plugin ":opensearch-sql-plugin"
-    keystore 'plugins.query.federation.catalog.config', new File("$projectDir/src/test/resources/catalog/", 'catalog.json')
+    keystore 'plugins.query.federation.datasources.config', new File("$projectDir/src/test/resources/catalog/", 'catalog.json')
```
Comment (YANG-DB) [inline integ-test/build.gradle:114]: can the catalog.json also be changed into datasource.json ?

**Review by ps48 — APPROVED** [review]

File: docs/user/general/identifiers.rst (hunk 1 above)
Comment (vamsi-amazon) [reply docs/user/general/identifiers.rst:234]: This is a backport PR can't do much. This is a good catch will fix it in coming PR.
The idea of this PR is to remove only public references[Documentation and show catlalogs command only] as it was done on the last day of release.

All the references in the implementation will be removed in a new PR.

**Pull request merged**